index.d.ts

```diff
@@ -0,0 +1,25 @@
+interface CodeReturn<T> {
+  width: number,
+  height: number,
+  data: T
+}
+
+type BufferRet = CodeReturn<Buffer>;
+type UintArrRet = CodeReturn<Buffer>;
+
+export declare function encode(imgData: Buffer, qu: number): BufferRet;
+
+/**
+ * @deprecated - decode takes an object since 0.3.5
+ */
+export declare function decode(jpegData: Buffer, opts: true): UintArrRet;
```

Comment:
FWIW it can accept anything `BufferLike` 

`Buffer | Uint8Array | number[]` should all work AFAIK, same goes for the encoding param